global-files/.gitattributes

```diff
@@ -0,0 +1,61 @@
+# Handle line endings automatically for files detected as text
+# and leave all files detected as binary untouched.
+* text=auto
+
+# Force the following filetypes to have unix eols, so Windows does not break them
+*.* text eol=lf
+
+# Windows forced line-endings
+/.idea/* text eol=crlf
```

Comment:
Is there a reason why we need to mention IDE specific files at all? I understand the `*.sln` and similar MS IDE files which today are also used by their `dotnet` CLI, but IntelliJ config files shouldn't be necessary anywhere.

Reply:
Because contributors and maintainers will use these tools and if we don't include them, their tools will break.

Reply:
Would this make the files in the `.idea` directory on Linux/macOS also have CRLF line endings?

Reply:
In my opinion IDE specific files should be ignored globally and shouoldn't be in any repository at all.

Unless you enforce a specific IDE at least project wide. As far as I know, such an enforcement only exists for the Delphi, Pharo and Elisp tracks. And here the enforcement is more or less given by the nature of the languages, that no standalone compilers exist.

Reply:
> In my opinion IDE specific files should be ignored globally and shouoldn't be in any repository at all.

And I don't agree. Unfortunately you need to know that you need to configure something in order for a repository to not break. And _many_ people don't actually know how to configure something.

On top of that, it's _almost impossible_ to override `.gitattributes` for "some globs", so in this case leaving editor configurations out will actively *break* people's environments.

Reply:
How should a git repository break things that are not intended to be part of the repository?

It is some IDE specific stuff, that should never be committed. So git will never temper with its line endings.

So how will removing this break peoples stuff?

Reply:
Because `.idea` isn't ignored _as a folder_, nor is `.sln`. So their repo will break, and can only unbreak if they now manually add it to the ignore. 

Which we can't do exercism-wide, because repos that require IntelliJ or Visual Studio (not code) would require some of the files in those directories to be commited.

I absolutely agree with you that in an ideal world it would be not in here at all.

Reply:
Again, I expect IDE files to be ignored on the users host, or not to be commited at all. I accept that there are `*.sln` files, for the dotnet tracks, its just how their tooling works, though there is not a single track that requires `.idea` beeing in the repository. So why should it ever be commited?

For erlang I would not accept a PR containing them. And if accidental commit breaks the committers IDE, I do not care. It was their fault to commit files that weren't intended for comitting.

A `.idea/` in a global git-ignore file, will ignore the folder and anything within it. Adding this to a global ignore file would be the solution for the problem.

Reply:
And if there will ever be a track requiring `.idea` in their repository, they still can add the necessary attributes to their repo-append.